### PR TITLE
Problem: ApplyMessageWithConfig parameter list is messy

### PR DESCRIPTION
**Pull request by yihuang**
Closes: #XXX

## Description

______

For contributor use:

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added  relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer

______

For admin use:

- [ ] Added appropriate labels to PR (ex. `WIP`, `R4R`, `docs`, etc)
- [ ] Reviewers assigned
- [ ] Squashed all commits, uses message "Merge pull request #XYZ: [title]" ([coding standards](https://github.com/tendermint/coding/blob/master/README.md#merging-a-pr))

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -84,6 +84,7 @@ Ref: https://keepachangelog.com/en/1.0.0/
 - (deps) [#381](https://github.com/crypto-org-chain/ethermint/pull/381) Upgrade Go-Ethereum version to [`v1.11.6`](https://github.com/ethereum/go-ethereum/releases/tag/v1.11.6).
 - (precompile) [#380](https://github.com/crypto-org-chain/ethermint/pull/380) Allow init precompiled contract with rules when new evm.
 - (precompile) [#383](https://github.com/crypto-org-chain/ethermint/pull/383) Allow init precompiled contract with ctx.
+- (evm) [#]() Cleanup ApplyMessageWithConfig interface.
```
Comment (yihuang) [inline CHANGELOG.md]:
```suggestion
- (evm) [#393](https://github.com/crypto-org-chain/ethermint/pull/393) Cleanup ApplyMessageWithConfig interface.
```

File: x/evm/keeper/grpc_query.go [hunk 2]
```diff
@@ -675,19 +670,20 @@ func (k *Keeper) traceMsg(
...
 
-	var stateOverrides rpctypes.StateOverride
 	if traceConfig.StateOverrides != nil {
 		config, err := json.Marshal(traceConfig.StateOverrides)
 		if err != nil {
 			return nil, 0, status.Error(codes.InvalidArgument, err.Error())
 		}
 
-		if err := json.Unmarshal(config, &stateOverrides); err != nil {
+		if err := json.Unmarshal(config, cfg.Overrides); err != nil {
```
Comment (mmsqe) [inline x/evm/keeper/grpc_query.go]:
```suggestion
		if cfg.Overrides == nil {
			cfg.Overrides = new(rpctypes.StateOverride)
		}
		if err := json.Unmarshal(config, cfg.Overrides); err != nil {
```

Comment (yihuang) [reply x/evm/keeper/grpc_query.go]: `cfg.Overrides` is a pointer already.

Comment (yihuang) [reply x/evm/keeper/grpc_query.go]: actually why we have json marshaling and unmarshalling at the same place, maybe you can fix that in your pr.

Comment (mmsqe) [reply x/evm/keeper/grpc_query.go]: it's pointer of pointer, no need init, but marshaling and unmarshalling here might just to ensure marshal no error?

Comment (yihuang) [reply x/evm/keeper/grpc_query.go]: I think it's simply to convert protobuf msg to custom struct

Comment (mmsqe) [reply x/evm/keeper/grpc_query.go]: seems need init Overrides

Comment (yihuang) [reply x/evm/keeper/grpc_query.go]: should be fixed now

**Review by mmsqe — APPROVED** [review]

**Pull request merged**